org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -38,6 +40,18 @@
 	 * Specifies the folder path to the JDK .
 	 */
 	public static final String JAVA_HOME = "java.home";
+	/**
+	 * Specifies the file path to the formatter xml url.
+	 */
+	public static final String JAVA_FORMATTER_URL = "java.formatter.url";
+	/**
+	 * Specifies the formatter profile name.
+	 */
+	public static final String JAVA_FORMATTER_PROFILE_NAME = "java.formatter.profileName";
+	/**
+	 * Preference key used to include the comments during the formatting.
+	 */
+	public static final String JAVA_FORMAT_COMMENTS = "java.format.comments";
 	/**
```

Comment:
I'm also working on ontype formatting so, in order to get a consistent naming scheme, I propose we use:

- java.format.settings.url
- java.format.setting.profile
- java.format.comments.enabled

Reply:
> java.format.settings.url
> java.format.setting.profile

settings or setting?

Reply:
settings. I fixed the typo after commenting